log/logger_setup.go

```diff
@@ -213,9 +217,10 @@ func registerNewSubLogger(subLogger string) *SubLogger {
 	}
 
 	temp := &SubLogger{
-		name:   strings.ToUpper(subLogger),
-		output: tempHolder,
-		levels: splitLevel("INFO|WARN|DEBUG|ERROR"),
+		name:    strings.ToUpper(subLogger),
+		output:  tempHolder,
+		levels:  splitLevel("INFO|WARN|DEBUG|ERROR"),
+		botName: logger.botName,
 	}
```

Comment:
I know I'm being a little annoying here, but with the ability to register new subloggers outside of `init` and the only time `structuredLogging` can be set to true is inside `SetupGlobalLogger`, would you be able to do something like check if the config has been populated and if so, set `structuredLogging` for this sublogger?

It will just prevent having some subloggers with and without structured logging